sdk/src/wallet/core/builder.rs

```diff
@@ -178,46 +175,102 @@ where
             self.secret_manager = secret_manager;
         }
 
-        let restored_bip_path = loaded_wallet_builder.as_ref().and_then(|builder| builder.bip_path);
+        let mut skip_verification = true;
```

Comment:
I'd personally flip the bool and call is `verify_*` since you end up negating it in the condition anyway, reads better IMO